src/lines-component.coffee

```diff
@@ -270,8 +270,19 @@ class LinesComponent
   remeasureCharacterWidths: ->
     return unless @presenter.baseCharacterWidth
 
-    @clearScopedCharWidths()
-    @measureCharactersInNewLines()
+    @measureCharactersInVisibleLines()
+
+  measureCharactersInVisibleLines: ->
+    for id, state of @oldState.lines
+      lineNode   = @lineNodesByLineId[id]
+      characters = lineNode.querySelectorAll("span.char")
+      charWidths = []
+
+      for character in characters
```

Comment:
Take a look at the previous code for an approach to measurement that doesn't depend on spans. You can build an iterator over the text nodes then set the start and end index of a DOM selection for a given pair of offsets in a text node to get a bounding rectangle from that.